src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11ECKeyFactory.java

```diff
@@ -294,7 +294,13 @@ <T extends KeySpec> T implGetPublicKeySpec(P11Key key, Class<T> keySpec,
             try {
                 token.p11.C_GetAttributeValue(session[0].id(), keyID, attributes);
                 ECParameterSpec params = decodeParameters(attributes[1].getByteArray());
-                ECPoint point = decodePoint(attributes[0].getByteArray(), params.getCurve());
+                ECPoint point;
+
+                if (!token.config.getUseEcX963Encoding()) {
+                    point = decodePoint(new DerValue(attributes[0].getByteArray()).getOctetString(), params.getCurve());
+                } else {
+                    point = decodePoint(attributes[0].getByteArray(), params.getCurve());
+                }
```

Comment:
Suggest inverting this to remove the negation.

Reply:
This code is a backport and is the exact code used in [openjdk](https://github.com/openjdk/jdk/blob/638d612c6b7c08c1f7be0d4e75e9f8a6dca1ef19/src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11ECKeyFactory.java#L291). I do not think it should be changed.

Reply:
Agreed.

Reply:
Was the Oracle copyright updated as part of the openjdk code backport?

Reply:
no it's still 2006, 2021. Should this be changed?

Reply:
I think Peter is saying that Oracle copyright date should be taken from the version that contained the change that is back-ported here. It looks to me like this particular change was made in 2020 so the Oracle date should remain 2021.